Allows tick events to auto-requeue when registered as listeners

cppcooper opened the pull request.

myk002 reviewed (commented):
after this change, what will the options be for cpp and lua code to handle regular callbacks? will we be able to replace `repeat-util` with `eventful`?

cppcooper commented on library/include/modules/EventManager.h:
```diff
@@ -144,6 +146,7 @@ namespace std {
             size_t r = 17;
             const size_t m = 65537;
             r = m*(r+(intptr_t)h.eventHandler);
+            r = m*(r+h.when);
             r = m*(r+h.freq);
```
Excellent point I had no considered.

cppcooper commented:
> after this change, what will the options be for cpp and lua code to handle regular callbacks? will we be able to replace `repeat-util` with `eventful`?

It looks as though that script could easily be replaced, yes.

cppcooper commented:
> after this change, what will the options be for cpp and lua code to handle regular callbacks? will we be able to replace `repeat-util` with `eventful`?

As discussed on discord, there are a few technical challenges when bringing tick events to lua. So I reverted those changes.

I've included an updated channel-safely in this PR which uses the newly repeating tick events to perform some tasks. Additionally I had added a temporary event which repeated hourly and performed a test to ensure the tick listener events are refiring without any user effort. [test was successful]
